extension/src/server/GradleServer.ts

```diff
@@ -29,65 +32,84 @@ export class GradleServer {
     constructor(
         private readonly opts: ServerOptions,
         private readonly context: vscode.ExtensionContext,
-        private readonly logger: Logger
+        private readonly logger: Logger,
+        private bspProxy: BspProxy
     ) {}
 
     public async start(): Promise<void> {
-        if (isDebuggingServer()) {
-            this.port = 8887;
-            this.fireOnStart();
-        } else {
-            this.port = await getPort();
-            const cwd = this.context.asAbsolutePath("lib");
-            const cmd = path.join(cwd, getGradleServerCommand());
-            const env = await getGradleServerEnv();
-            if (!env) {
+        this.taskServerPort = await getPort();
+        const cwd = this.context.asAbsolutePath("lib");
+        const cmd = path.join(cwd, getGradleServerCommand());
+        const env = await getGradleServerEnv();
+        const bundleDirectory = this.context.asAbsolutePath("server");
+        if (!env) {
+            await vscode.window.showErrorMessage(NO_JAVA_EXECUTABLE);
+            return;
+        }
+        let startBuildServer = "false";
+        if (redHatJavaInstalled()) {
+            const javaExecPath = getRedHatJavaExecutablePath() || (await getJavaExecutablePath());
+            if (!javaExecPath) {
                 await vscode.window.showErrorMessage(NO_JAVA_EXECUTABLE);
-                return;
+            } else {
+                startBuildServer = "true";
             }
-            const args = [String(this.port)];
-
-            this.logger.debug("Starting server");
-            this.logger.debug(`Gradle Server cmd: ${cmd} ${args.join(" ")}`);
-
-            this.process = cp.spawn(`"${cmd}"`, args, {
-                cwd,
-                env,
-                shell: true,
-            });
-            this.process.stdout.on("data", this.logOutput);
-            this.process.stderr.on("data", this.logOutput);
-            this.process
-                .on("error", (err: Error) => this.logger.error(err.message))
-                .on("exit", async (code) => {
-                    this.logger.warn("Gradle server stopped");
-                    this._onDidStop.fire(null);
-                    this.ready = false;
-                    this.process?.removeAllListeners();
-                    if (this.restarting) {
-                        this.restarting = false;
-                        await this.start();
-                    } else if (code !== 0) {
-                        await this.handleServerStartError();
-                    }
-                });
-
-            this.fireOnStart();
         }
+        const buildServerPipeName = this.bspProxy.getBuildServerPipeName();
+        const args = [
+            `--port=${this.taskServerPort}`,
+            `--pipeName=${buildServerPipeName}`,
+            `--bundleDir=${bundleDirectory}`,
+            `--startBuildServer=${startBuildServer}`,
+        ];
```

Comment:
It don't need to send `bundleDirectory` and `buildServerPipeName` if `startBuildServer == false`, will fix it